src/inc-lib-variants.php

```diff
@@ -62,11 +70,13 @@ function lovd_fixHGVS ($sVariant, $sType = 'g', $sReference = '')
     // Check for a reference sequence.
     if ($sReference !== false) {
         if ($sReference == '') {
-            if (preg_match('/^[NX][CGMR]_[0-9]{6,9}\.[0-9]+:/', $sVariant)) {
+            if (preg_match('/^(\(?(LRG|ENS[GT]|[NX][CGMRTW])_?[0-9]+([.t][0-9]+)?\)?){1,2}:/', $sVariant)) {
                 list($sReference, $sVariant) = explode(':', $sVariant);
-                $sReference .= ':'; // To easy the concatenation.
+                $sReference .= ':'; // We add the ':' to ease the concatenation later on.
             } else {
-                // No reference was found.
+                // No reference was found. We fill this in by false so
+                //  the function will not have to continue to do this
+                //  test every time the function calls upon itself.
```

Comment:
Looking at the code, I didn't think the function could still call itself well, so I tried to add a test for it. But I keep getting weird results and even an infinite loop. I have to investigate, I'm not sure what's going on.